### ISSUE TITLE
[BUG] When reviewing an entire note, the metadata is applied incorrectly (on top of the already existing metadata)

### ISSUE THREAD

**Issue by SubZeroX** [st3v3nmw/obsidian-spaced-repetition#776]
**Describe the bug**
When reviewing an entire note, the metadata is applied incorrectly (on top of the already existing metadata)

**To Reproduce**

1. Choose a note
2. Use any hotkey to revise the note
3. See error

**Expected behavior**
.

**Screenshots**

https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/13406988/82a2c762-d9af-4766-9d08-eeb0b2b3facb

**Versions (please complete the following information):**

-   OS: Windows 10
-   Obsidian version: 1.4.16
-   Plugin version: latest
-   If on desktop, Installer version: 5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) obsidian/1.4.13 Chrome/114.0.5735.289 Electron/25.8.1 Safari/537.36

**Comment by ronzulu** [comment]
Hi  @SubZeroX 

Which plugin version are you using?

Also, if you could please include the text of the Metadata you showed in the video, that would be helpful.

Cheers
Ronny

**Comment by SubZeroX** [comment]
Hi @ronzulu 

I'm using 1.10.3.

The metadata is: 
```yaml
created: 19-10-2023
type: PermanentNote
alias: [""]
tag: ["#PUClínica"]
```

Thanks!

**Comment by ronzulu** [comment]
Hey  @SubZeroX  thanks for that.

Unfortunately I can't reproduce the problem. Could you please attach the actual md file. Also, does this consistently happen, or only sometimes?

I see that you are also using Windows 10 like myself, so at least we can rule out OS differences as been the cause.

And similar to https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/764, where as yet I haven't been able to reproduce.

Cheers
Ronny

**Comment by ronzulu** [comment]
Hi @SubZeroX thanks for all the help on diagnosing https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/764.

Is this issue #766 still an issue, or did the fix for https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/764 fix this as well?

**Comment by SubZeroX** [comment]
Hey @ronzulu, I was waiting a new version to be release so I can keep on exploring this bug. Actually, it still happens, I'll record a video tomorrow and explain it better.

**Comment by ronzulu** [comment]
@SubZeroX and just checking, we are still talking about the interim version of `main.js` from:
https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/764#issuecomment-1788890836

I didn't want to place a pull request on the fix for #764 until I knew about this one.

Cheers
Ronny

**Comment by SubZeroX** [comment]
Hey @ronzulu, 

> and just checking, we are still talking about the interim version of main.js 

Yes.

https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/13406988/303bb61b-758c-44fc-b7c8-7125d69e720f

So as you can see the problem is intermittent but happens quite frequently.

Here is the note: 

[Teste Ergométrico.md](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/13260056/Teste.Ergometrico.md)


And here is the printscreen: 
![Obsidian_28EffoNoYS](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/13406988/461c9acd-0cb5-4097-b575-8ce83c5edfbc)

**Comment by ronzulu** [comment]
Hey @SubZeroX love the soundtrack!

The issue seems the same as https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/647, which is before my contributions.

I'll have a little look and see if I can fix.

Cheers
Ronny

**Comment by ronzulu** [comment]
Hi again @SubZeroX

I couldn't reproduce, so I've added some debug that should narrow down the problem.

[main.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/13263231/main.zip)

e.g.:
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/22d01f94-9f62-4223-b1bf-9f16f617a702)

Cheers
Ronny

**Comment by SubZeroX** [comment]
Hey @ronzulu, here is the screenshot

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/13406988/efa06859-aefa-4928-bb4c-65bbf1c5b2f4)

**Comment by ronzulu** [comment]
Hiya  @SubZeroX, that was helpful.

The existing code isn't picking up that there is frontmatter. So it inserts the OSR YAML at the start of the file (like it would if there was no frontmatter) and tacking on the entire original file.

I've added some extra debugging to show in hex the content of the file (first 500 characters). That should give me everything I need to fix the problem.

Screenshot of the following would be great:

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/4e866786-e27a-4541-bacf-7a232d7cdc55)

Cheers
Ronny

**Comment by SubZeroX** [comment]
Hey @ronzulu, I don't quite know what happened, but the file you attached doesn't stop uploading and therefore I can't download it

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/13406988/26b872ae-480e-4e65-867d-c9a8cc919fe6)

**Comment by ronzulu** [comment]
@SubZeroX 
Strange, don't know what happened... hopefully works this time

[main_776.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/13296696/main_776.zip)

**Comment by SubZeroX** [comment]
Hey @ronzulu,

Here is the screenshot
![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/13406988/80113c0e-da79-4b1e-8a0d-d326fafe4b16)

**Comment by ronzulu** [comment]
Hi  @SubZeroX 

Are you able to check out this version... if it doesn't work please take a screenshot of the chrome developer console like before.

[main_776C.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/13494860/main_776C.zip)

Cheers
Ronny

**Comment by SubZeroX** [comment]
Hey @ronzulu 

It's working 100%. Thanks once again for your time and effort in the developing of this amazing plugin.

**Comment by ronzulu** [comment]
Glad to know it works (and it wasn't a bug I created!). 

I've opened the merge request, so hopefully makes it into the release version in the next couple of weeks or so.

**Comment by ronzulu** [comment]
Hi @SubZeroX the fix has been included in the latest release 1.11.0. Please close this at your convenience.

Cheers
Ronny
